esphome/components/sx126x/__init__.py

```diff
@@ -201,7 +201,9 @@ def validate_config(config):
             ),
             cv.Optional(CONF_DEVIATION, default=5000): cv.int_range(min=0, max=100000),
             cv.Required(CONF_DIO1_PIN): pins.gpio_input_pin_schema,
-            cv.Required(CONF_FREQUENCY): cv.int_range(min=137000000, max=1020000000),
+            cv.Required(CONF_FREQUENCY): cv.All(
+                cv.frequency, cv.float_range(min=137000000, max=1020000000)
```

Comment:
Maybe use scientific notation so not trying to count the zeros, or better still update `cv.frequency` to take min and max values in formats like "137MHz".

```suggestion
                cv.frequency, cv.float_range(min=137.0e6, max=1.02e9)
```